controlplane/kubeadm/internal/filters.go

```diff
@@ -34,9 +34,9 @@ import (
 // Kubernetes version, infrastructure template, and KubeadmConfig field need to be equivalent.
 func MatchesMachineSpec(infraConfigs map[string]*unstructured.Unstructured, machineConfigs map[string]*bootstrapv1.KubeadmConfig, kcp *controlplanev1.KubeadmControlPlane) func(machine *clusterv1.Machine) bool {
 	return collections.And(
-		func(machine *clusterv1.Machine) bool {
-			return matchMachineTemplateMetadata(kcp, machine)
-		},
+		//func(machine *clusterv1.Machine) bool {
```

Comment:
I think we still have to trigger a rollout if the cluster config annotation changed.

We use this annotation to store with which configuration we ran kubeadm init/join on the machine.

I think we can't simply skip rollout by in-place mutating this annotation

```go
	// Machine's bootstrap config may be missing ClusterConfiguration if it is not the first machine in the control plane.
	// We store ClusterConfiguration as annotation here to detect any changes in KCP ClusterConfiguration and rollout the machine if any.
```

Reply:
That is covered under `MatchesKubeadmBootstrapConfig`. We dont need to do anything else here.

Reply:
Ah you're right!